### PR TITLE
[CFP-352] FormBuilder migration - Hardship fees

### PR DESCRIPTION
**Pull request by imtiazAhmd**
#### What

Migrate external_users form step: **hardship_fees** form

#### Ticket

[FormBuilder migration - Hardship fees](https://dsdmoj.atlassian.net/browse/CFP-352)

#### Why

- Update the service to use GOV.UK styles, components and patterns
- To improve and resolve the service's accessibility issues
- To reduce code complexity
- To improve maintainability

--------

#### TODO (wip)

 - [x] Review views
 - [x] Review validators
 - [x] Review locales
 - [x] Review specs

### CODE REVIEW

**Review by mpw5 — APPROVED** [review]
LGTM, but I'd get a front-end developer to 👀 it too 🎉

File: app/views/external_users/claims/hardship_fee/_fields.html.haml [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
-        = hf.adp_text_field :amount,
-          label: t('.amount'),
-          input_classes: 'fee-amount form-control-1-4 form-input-denote__input total',
-          input_type: 'currency',
-          value: number_with_precision(hf.object.amount, precision: 2),
-          id: 'hardship_fee_attributes_amount',
-          errors: @error_presenter,
-          error_key:'hardship_fee.amount'
+      = hf.govuk_number_field :quantity,
+        class:'quantity fee-quantity js-fee-calculator-ppe form-control-1-4',
```
Comment (naseberry) [inline app/views/external_users/claims/hardship_fee/_fields.html.haml]: CSS class `form-control-1-4` is redundant

File: config/locales/en/error_messages/claim.yml [hunk 2]
```diff
@@ -1939,46 +1939,50 @@ interim_fee:
...
   fee_type:
     _seq: 10
-    blank:
+    choose_a_type_for_the_hardship_fee:
       long: Choose a type for the hardship fee
-      short: Choose a fee type
+      short: _
       api: Choose a type for the hardship fee
 
   quantity:
```
Comment (naseberry) [inline config/locales/en/error_messages/claim.yml:1961]: Should `quantity` have a key:value pair for the attribute `:item_max_amount`
The validator `def validate_presence_and_numericality_govuk_formbuilder` validates against `:blank`, `:numericality` and `:item_max_amount`

File: config/locales/en/models/claim.yml [hunk 3]
```diff
@@ -262,6 +262,21 @@ en:
...
             quantity:
               numericality: Enter a valid quantity for the graduated fee
+        fee/hardship_fee:
+          attributes:
+            blank: Add a hardship fee
+            base:
+              uneditable_state: You cannot edit a claim that is not in draft state
+            fee_type:
+              blank: Choose a type for the hardship fee
+            quantity:
```
Comment (naseberry) [inline config/locales/en/models/claim.yml:272]: Should `quantity` have a key:value pair for the attribute `:item_max_amount`
The validator `def validate_presence_and_numericality_govuk_formbuilder` validates against `:blank`, `:numericality` and `:item_max_amount`

Comment (imtiazAhmd) [reply config/locales/en/models/claim.yml:272]: this key:value already exists in later commit.

File: config/locales/en/error_messages/claim.yml (hunk 2 above)
Comment (imtiazAhmd) [reply config/locales/en/error_messages/claim.yml:1961]: Done in latest commit

File: app/views/external_users/claims/hardship_fee/_fields.html.haml (hunk 1 above)
Comment (imtiazAhmd) [reply app/views/external_users/claims/hardship_fee/_fields.html.haml]: Done in latest commit

**Review by naseberry — APPROVED** [review]
:technologist: LGTM

**Pull request merged**